docs/src/use/configure/configuration-files.md

```diff
@@ -318,6 +319,32 @@ Here, the `js.configs.recommended` predefined configuration is applied first and
 
 For more information on how to combine predefined configs with your preferences, please see [Combine Configs](combine-configs).
 
+### Convention for Configuration Name
+
+The `name` property is optional, but it is recommended to provide a name for each configuration object, especially when you are creating shared configurations. The name is used in error messages and the config inspector to help identify which configuration object is being used.
+
+We recommend using a name that is descriptive of the configuration object's purpose and scoped with the configurations name or plugin name. Using `/` as the seperator. For example, if you are creating a configuration object for a plugin named `eslint-config-example`, you might name the configuration object `example/recommended`.
+
+```js
+export const recommanded = [
+    {
+        name: "example/recommended",
+        rules: {
+            "no-unused-vars": "warn"
+        }
+    }
+]
+
+export const strict = [
+    {
+        name: "example/strict",
+        rules: {
+            "no-unused-vars": "error"
+        }
+    }
+]
```

Comment:
Is this supposed to be a plugin? If so, we should use the same format we use in other plugin docs:

https://eslint.org/docs/next/extend/plugins#configs-in-plugins

Reply:
It seems we don't have a clear convention for the flat config style configures packages.

For example, `@eslint/js` exports a single configure object with default export:

```ts
import eslint from '@eslint/js`
```

While in some cases, `eslint-config-example` might want to expose multiple configs for different usage, should one use named exports, or nested in a single configs object?

```ts
import { recommended, strict } from 'eslint-config-example'
// or
import configs from 'eslint-config-example'
```

We also have cases like `typescript-eslint` that exposes a factory function (https://typescript-eslint.io/getting-started)

I am not sure if it's something we want to "control" or just let each package to decide, since it's just JavaScript now.

Here I was trying to demonstrate the second case - but I am up to change to any format that make more sense.

Reply:
People can do whatever they want, that's the fun of flat config. :)

There is an issue open here to come up with recommendations:
https://github.com/eslint/eslint/issues/18095

I just think that if we are mentioning naming configs in plugins, then the code example should match other plugin documentation.